src/Abstractions/NexusMods.Abstractions.Loadouts.Synchronizers/ALoadoutSynchronizer.cs

```diff
@@ -239,8 +242,7 @@ await foreach (var entry in _hashCache.IndexFolderAsync(location))
 
 
         // Extract all the files that need extracting in one batch.
-        await _fileStore.ExtractFiles(toExtract
-            .Select(f => (f.Value.Hash, f.Key)));
```

Comment:
Was this really bottlenecking?
Do you have data on the improvements for using the new function for curiosity?

Reply:
It's complicated to state the improvement here in isolation.

Normally something like this alone won't be a bottleneck, what was a bottleneck was taking this `IEnumerable` and then doing multiple LINQ operations on that enumerable inside the `ExtractFiles` method, creating a much bigger enumerable.

And then with that enumerable, making singlethreaded DB calls per file to obtain the archive in which each file is stored.

That logic was parallelized with around ~5x improvement if I recall. Mostly due to parallel DB calls.

For the parallel logic, knowing the size (length) of the collection was also handy for the purposes of preallocating. So it's hard to narrow how much this specific line in isolation changes. It was part of a bigger set of patches.

Reply:
If I were to have a completely random guess though, it's probably ~1.6x as fast, and with less allocations. But it won't amount into much, probably like 20-40 millis improvement on a 100K file input on this line.

I just did it as a I de-LINQ'ed the whole codepath, as the logic inside the `ExtractFiles` which used the enumerable was originally bottlenecking.